### DISCUSSION TITLE
After load event on recommend-js?

### DISCUSSION THREAD

**Discussion by andrewwise** [algolia/react-recommendations#138] [Q&A] [answered]
Is there any kind of after load event when, for example, `relatedProducts()` is fully loaded?

I've got a set up like this at the moment to initialise a third-party slider on the loaded products:

```js
fetch(
  relatedProducts({
    // config
  });
).then(function() {
  // init third-party slider
});
```

On certain browsers sometimes this seems to run the third-party slider init code before the products/html from `relatedProducts()` is fully loaded.

**Comment by raed667** [comment, marked as answer]
There are currently 2 ways to achieve this: 

- Using [InstantSearch.js](https://www.algolia.com/doc/guides/algolia-recommend/upgrade/recommend-js/) you can checkout this StackOverflow answer https://stackoverflow.com/questions/42672838/algolia-instantsearch-add-a-callback-after-results-are-displayed  

- Using the [Recommend API client](https://www.npmjs.com/package/@algolia/recommend) 

This is a lightweight API client that gives you back the recommendations as an object. This is the most versatile way to hook into before/after events. However, with this approach you'll need to handle rendering on your own.


I hope that helps !
